### DISCUSSION TITLE
All steps detailed?

### DISCUSSION THREAD

**Discussion by claireewilliams** [metagenome-atlas/atlas#514] [General]
Hi there! I'm really excited about this workflow. It is clearly a lot of work and simplifies the metagenomics workflows I've been exploring in a really user-friendly way. Great work! 

I was wondering, though, if there was a document somewhere that explicitly states all the tools that are being called through the process. For example, I can't seem to find whether, for the metatranscriptomics usage, rRNA is removed. And if so, what program is used? I was just hoping there was a list somewhere that said X, Y, Z are done in X order based on data type.

Thanks! 
Claire

**Comment by SilasK** [comment]
To your general question: There is this image, which shows the main tools:

![[Tools in atlas](https://github.com/metagenome-atlas/atlas/blob/master/resources/images/atlas_list.png)](https://github.com/metagenome-atlas/atlas/blob/master/resources/images/atlas_list.png)


Also, the [Docs](https://metagenome-atlas.readthedocs.io/en/latest/usage/output.html) give some details on the tools used. 
But maybe I could detail this a bit more. 


Now, it seems you are asking specifically about the metatranscriptomics workflow. Wile atlas supports metatranscriptomics since the beginning, I never had really data to test it, until now. I, therefore, plan to extend the pipeline on this front. 

I think what we have until now it the `atlas qc` which does the normal quality control using bbmap, **and it removes reads mapping to 16S** if you specify `metatranscriptome` as a datatype. 

But from then I'm not sure what to do with the reads. What would you do? assembly? do you have the metagenome linked to the same samples?

**Comment by LeeBergstrand** [comment]
@claireewilliams Run `atlas run all --dryrun --dag | dot -Tpdf > dag.pdf` it will make a PDF that explains all the steps being performed and how they rely on each other. Though the steps (e.g. rules) don't have programs associated with them you can look up the rule names in the atlas code. 

![image](https://user-images.githubusercontent.com/5819462/168689584-aacf2095-5b08-4ad2-bf43-77acc907adad.png)

**Reply by SilasK** [reply]
If you ran atlas, at the end you can produce a `--report report.html` there you have this graphic and all the details.

**Reply by s-junguy** [reply]
Is it possible only to generate this report without running all atlas steps again?

**Reply by SilasK** [reply]
If i am not mistaken the --dag doesn't execute the jobs. There is also a rulegraph option.

**Comment by raw937** [comment]
@LeeBergstrand What is the program that generates that plot above?

**Reply by LeeBergstrand** [reply]
`atlas run all --dryrun --dag | dot -Tpdf > dag.pdf` --> `--dag` dumps a DAG file to screen and `dot -Tpdf` makes the PDF of the DAG file.

**Reply by raw937** [reply]
I am wondering the general program that makes these workflow plots. Not how atlas does it. The name of the program that makes the DAG file.

**Reply by LeeBergstrand** [reply]
Because ATLAS uses snakemake as its execution engine, any command-line flags after the `atlas run all` are snakemake flags. You can use almost any command flag in the snakemake CLI (https://snakemake.readthedocs.io/en/stable/executing/cli.html) with ATLAS. The creation of the DAG file is a built-in feature of snakemake and can be accessed by adding the `--dag` flag to any snakemake workflow. See here for details: https://snakemake.readthedocs.io/en/stable/executing/cli.html#visualization. With regards to the underlying code that snakemake uses to generate the DAG file, I'm am unsure if its built-in or from a library that one could use.

**Reply by SilasK** [reply]
I think it's graphviz. For info. There is also a D3 option in snake make.
